lib/client-assets.php

```diff
@@ -866,7 +866,7 @@ function gutenberg_editor_scripts_and_styles( $hook ) {
 	// Prepare Jed locale data.
 	$locale_data = gutenberg_get_jed_locale_data( 'gutenberg' );
 	wp_add_inline_script(
-		'wp-edit-post',
+		'wp-components',
```

Comment:
I think this doesn't solve the conceptual problem but just moves it to another level. I think the solution here would be to create a separate script for these i18n strings and add it as a dependency to all Gutenberg scripts with translatable strings. I used this technique here https://github.com/youknowriad/gcf/pull/33/files#diff-14f2377adcf109755a6ae6321d6c35afR1

Reply:
Why so complicated if you can just use wp-i18n?

Reply:
Because if you use `wp-i18n` you're adding this script to all pages where maybe it's not needed. I guess the ideal is a domain per script :) to avoid loading everything on all pages.

I can be on board with `wp-i18n` for Gutenberg (because it's Core) but not for plugins though. I also wonder how this would grow over time :).

Reply:
Mhh, you only enqueue and add inline scripts when the page needs them. Just because plugin X adds the translations for page Y, it doesn’t mean that the translations are automatically available on all pages.

Reply:
It doesn't make sense to load the `wp-components` scripts in any page without the translated strings it depends on.

It doesn't make sense to load the `wp-editor` scripts in any page without the translated strings it depends on.

If a plugin author creates a new page in WP Admin and is depending on `wp-components` because it's using one of these generic components. The only thing this plugin author should do is add a dependency on `wp-components` to its scripts where these components are used, the plugin author shouldn't have to worry about setting translated strings for this module, this is just implementation detail of the `wp-components` script.

So this means the translated strings are just a dependency for the `wp-components` like any other dependency and it should be defined at registration time and not when enqueuing the scripts.

If you decide to add this inline script at enqueue time, you'll have to do it on each page this script is loaded in.

Reply:
Again, that's the theory, I'm fine with making a compromise for Gutenberg translation strings (or Core JS translation strings in general) and just always load them in all pages. (which means adding them to `wp-i18n` at registration time)

Reply:
I'm not really following you logic here.
`wp-components`, `wp-blocks`, `wp-editor` and `wp-edit-post` are currently the only scripts that depend on `wp-i18n` . Since the strings are not separated and based on the dependency tree, `wp-components` is the common script used by all of them and therefore should be used to register the translations. Just like in this PR demonstrated.

How scripts and translations are registered with Gutenberg in core is another topic and should probably be part of the merge proposal.

Reply:
> `wp-components` is the common script used by all of them and therefore should be used to register the translations. Just like in this PR demonstrated.

That's true now but that's just a coincidence, what if we introduce a new Gutenberg module that doesn't depend on components but still requires i18n, what if we add a translated string to `wp-data` in another PR? I'd be more confident (more future proof) if we add this as to dependency of `wp-i18n`